awx/api/fields.py

```diff
@@ -102,7 +102,8 @@ def __init__(self, **kwargs):
 
     def to_internal_value(self, pk):
         try:
+            pk = int(pk)
             Credential.objects.get(pk=pk)
-        except ObjectDoesNotExist:
+        except (ObjectDoesNotExist, ValueError):
             raise serializers.ValidationError(_('Credential {} does not exist').format(pk))
```

Comment:
The base class from the Django rest framework handles this much more elegantly.

https://github.com/encode/django-rest-framework/blob/9d06e43d05abf1ec57f15566b29ad53ac418ae05/rest_framework/fields.py#L923-L958

looks like it would give the error

> A valid integer is required.

it looks like you might actually be able to get this by deferring to the super class before the existing `try` block.